compiler/src/dotty/tools/dotc/sbt/ExtractAPI.scala

```diff
@@ -620,27 +620,128 @@ private class ExtractAPICollector(using Context) extends ThunkHolder {
    *  it should stay the same across compiler runs, compiler instances,
    *  JVMs, etc.
    */
-  def treeHash(tree: Tree): Int =
+  def treeHash(tree: Tree, inlineSym: Symbol): Int =
     import scala.util.hashing.MurmurHash3
+    import core.Constants.*
+
+    val seenInlines = mutable.HashSet.empty[Symbol]
+
+    if inlineSym ne NoSymbol then
+      seenInlines += inlineSym // do not hash twice a recursive def
```

Comment:
Why do we need to do `seenInlines += ...` both here and in positionedHash? Can they both loop independently?

Reply:
Its to prevent redundant double hashing the parameters and return type of a recursive inline method